### PR TITLE
Added docs about playbooks

### LINKED ISSUE #292: 🚀 Feature: Add documentation for the playbook fields
### 🔖 Feature description

Refer to the `app/engines/axios.js` and create a clear documentation for non-programmers' reference and to help them in contributing to the project. 

### 🎤 Pitch

The documentation should be in as simple words as possible so that a person with close to zero programming experience can also contribute to the project.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar feature request

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by akgupta0777**
# Description

This is my first attempt at writing docs so It's open for improvements. This PR introduces docs for playbooks fields.

Fixes #292 

## How Has This Been Tested

No test Needed

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation/readme
- [X] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by drph4nt0m** [comment]
I like the idea of having an example in the playbook documentation but I was hoping for more like a generic documentation. Maybe like a table with the keys as the first column and then have multiple columns for diff things like description, data type, required field, parent key, example, etc. And then have a small example below it. 

Check https://spec.openapis.org/oas/v3.1.0 for some ideas. This is used for API specifications but can be kinda used here also.

**Comment by akgupta0777** [comment]
> I like the idea of having an example in the playbook documentation but I was hoping for more like a generic documentation. Maybe like a table with the keys as the first column and then have multiple columns for diff things like description, data type, required field, parent key, example, etc. And then have a small example below it.
> 
> Check https://spec.openapis.org/oas/v3.1.0 for some ideas. This is used for API specifications but can be kinda used here also.

Hmm, that's a good idea. I'll try to implement this tomorrow.
will make different versions of docs in separate PRs for a clean experience. In the end which PR fits better will be merged.

**Comment by drph4nt0m** [comment]
You can work in the same PR. We can always go back to the old commit. Not an issue.

**Comment by akgupta0777** [comment]
friendly ping @drph4nt0m

File: PLAYBOOKS.md [hunk 1]
```diff
@@ -10,44 +10,21 @@ A playbook is a JSON file that will be used by our [scraping engine](./app/engin
...
+| source        | Source of information                                  | String           | Yes      | source     | `"source": "Airports Authority of India"`                                                                                        |
+| engine        | Name of scraping engine(Axios currently)               | String           | Yes      | scraper    | `"engine": "axios"`                                                                                                              |
+| baseUrl       | URL for charts website(eAIP)                           | String           | Yes      | features   | `"baseUrl": "https://aim-india.aai.aero/eaip-v2-02-2021"`                                                                        |
+| paths         | Path for getting charts from eAIP                      | Array            | Yes      | features   | [See example](https://github.com/drph4nt0m/avbot-charts/blob/ec58c14ae01da70bcff68a4360027a2142a80366/playbooks/AU.json#L11-L15) |
```
Comment (drph4nt0m) [inline PLAYBOOKS.md:20]: Path is not `Path for getting charts from eAIP` but it defines multiple routes it can go to

File: PLAYBOOKS.md [hunk 2]
```diff
@@ -10,44 +10,21 @@ A playbook is a JSON file that will be used by our [scraping engine](./app/engin
...
+| route         | Route for the front page                               | String           | Yes      | paths      | `"route": "/index-en-GB.html"`                                                                                                   |
```
Comment (drph4nt0m) [inline PLAYBOOKS.md:21]: route is not `Route for the front page` but it tells multiple routes it can go to under the baseUrl

File: PLAYBOOKS.md [hunk 3]
```diff
@@ -10,44 +10,21 @@ A playbook is a JSON file that will be used by our [scraping engine](./app/engin
...
+| navigations   | Navigations for the engine to get charts page          | Array            | Optional | paths      | [See example](https://github.com/drph4nt0m/avbot-charts/blob/ec58c14ae01da70bcff68a4360027a2142a80366/playbooks/IN.json#L14-L23) |
```
Comment (drph4nt0m) [inline PLAYBOOKS.md:22]: Selector based navigations

Comment (drph4nt0m) [reply PLAYBOOKS.md:22]: add the child keys of the navigations

File: PLAYBOOKS.md [hunk 4]
```diff
@@ -10,44 +10,21 @@ A playbook is a JSON file that will be used by our [scraping engine](./app/engin
...
+| chart.baseUrl | URL for charts page from eAIP                          | String           | Yes      | chart      | [See example](https://github.com/drph4nt0m/avbot-charts/blob/ec58c14ae01da70bcff68a4360027a2142a80366/playbooks/IN.json#L26-L30) |
```
Comment (drph4nt0m) [inline PLAYBOOKS.md:23]: Not exactly the url for charts page but just a prefix for the value we get from href using the selector

File: PLAYBOOKS.md [hunk 5]
```diff
@@ -10,44 +10,21 @@ A playbook is a JSON file that will be used by our [scraping engine](./app/engin
...
+| selector      | HTML element selector                                  | String or Regexp | Yes      | chart      | `"selector": "a[title=\"${icao}\"]"`                                                                                             |
+| attribute     | Attribute of HTML selector                             | String           | Yes      | chart      | `"attribute": "href"`                                                                                                            |
+| regex         | Regular Expression for finding selectors based on icao | Regexp           | Optional | chart      | `"regex": "\\(${icao}\\)" `                                                                                                      |
```
Comment (drph4nt0m) [inline PLAYBOOKS.md:26]: Regular expression for finding text (not selectors) in the body innerText

File: PLAYBOOKS.md (hunk 1 above)
Comment (drph4nt0m) [inline PLAYBOOKS.md:20]: paths is not `Path for getting charts from eAIP` but it defines multiple routes it can go to

File: PLAYBOOKS.md [hunk 6]
```diff
@@ -10,44 +10,21 @@ A playbook is a JSON file that will be used by our [scraping engine](./app/engin
...
+| name          | Name of the country                                    | String           | Yes      | Country    | `"name": "India"`                                                                                                                |
+| source        | Source of information                                  | String           | Yes      | source     | `"source": "Airports Authority of India"`                                                                                        |
+| engine        | Name of scraping engine(Axios currently)               | String           | Yes      | scraper    | `"engine": "axios"`                                                                                                              |
+| baseUrl       | URL for charts website(eAIP)                           | String           | Yes      | features   | `"baseUrl": "https://aim-india.aai.aero/eaip-v2-02-2021"`                                                                        |
```
Comment (drph4nt0m) [inline PLAYBOOKS.md:19]: Prefix of the URL that will be added before the routes in paths

**Review by drph4nt0m — CHANGES_REQUESTED** [review]

**Comment by drph4nt0m** [comment]
@akgupta0777 I made few changes on your initial idea. Would like your feedback also before merging. Think we should add something more?

**Review by akgupta0777 — COMMENTED** [review]
The markdown has some bug in a little area. Its not previewing correctly.

**Comment by akgupta0777** [comment]
image

Fixing this

**Comment by akgupta0777** [comment]
This is not bug in markdown. It's prettier which is breaking the markdown

**Comment by drph4nt0m** [comment]
I feel it is your local editor settings messing around with it. The repo's prettier config didn't break it for me.

**Comment by akgupta0777** [comment]
> I feel it is your local editor settings messing around with it. The repo's prettier config didn't break it for me.

Your code is also not previewing markdown. I fixed it and it was working but after prettier i again checked the preview is messed up and markdown is broken.

**Comment by akgupta0777** [comment]
@drph4nt0m  [See Github Preview here](https://github.com/drph4nt0m/avbot-charts/pull/296/files?short_path=ecbb7fd#diff-ecbb7fd51a415c8033e01d36b371d74fb1da28c3194c5ad0fa320c33805184aa)

File: PLAYBOOKS.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+| `scraper.features.chart`                           | The CSS selector based configuration for getting and formatting the final chart URL                               | JSON         | Yes (can either use `chart` or `search`)   | [See example](playbooks/IN.json)                          |
+| `scraper.features.chart.baseUrl`                   | Constant prefix part of the URL to be used for all chart URLs                                                     | String       | Yes                                        | `"baseUrl": "${baseUrl}/eAIP/"`                           |
+| `scraper.features.chart.selector`                  | CSS Selector for the element that is to be used for getting the final chart URL                                   | String       | Yes (can either use `selector` or `xpath`) | `"selector": "a[title=\"${icao}\"]"`                      |
+| `scraper.features.chart.xpath`                     | XPath of the                                                                                                      | String       | Yes (can either use `selector` or `xpath`) | `"xpath": "//a[contains(text(),'${icao}')]"`              |
```
Comment (akgupta0777) [inline PLAYBOOKS.md:36]: Xpath of the ... missing text afterwards.

**Review by akgupta0777 — COMMENTED** [review]
everything else seems good. just a minor issue

**Comment by drph4nt0m** [comment]
> @drph4nt0m [See Github Preview here](https://github.com/drph4nt0m/avbot-charts/pull/296/files?short_path=ecbb7fd#diff-ecbb7fd51a415c8033e01d36b371d74fb1da28c3194c5ad0fa320c33805184aa)

That is the git diff of the changes. And wont have proper view of the markdown. Manually go to `Files Changed` > three dots > View file (https://github.com/drph4nt0m/avbot-charts/blob/b9caf65c7be70c2b35849d6f63dbc31a0ddf9649/PLAYBOOKS.md)

**Comment by akgupta0777** [comment]
> > @drph4nt0m [See Github Preview here](https://github.com/drph4nt0m/avbot-charts/pull/296/files?short_path=ecbb7fd#diff-ecbb7fd51a415c8033e01d36b371d74fb1da28c3194c5ad0fa320c33805184aa)
> 
> That is the git diff of the changes. And wont have proper view of the markdown. Manually go to `Files Changed` > three dots > View file (https://github.com/drph4nt0m/avbot-charts/blob/b9caf65c7be70c2b35849d6f63dbc31a0ddf9649/PLAYBOOKS.md)

Even I tried this way too its still broken

**Comment by akgupta0777** [comment]
image

Looks like this

**Comment by drph4nt0m** [comment]
Looking into it

File: PLAYBOOKS.md (hunk 7 above)
Comment (drph4nt0m) [reply PLAYBOOKS.md]: Fixed

**Comment by akgupta0777** [comment]
How it's resolved

**Comment by drph4nt0m** [comment]
I feel it is fixed now. `scraper.features.paths / route` at line 30 was mistyped as `scraper.features.paths | route` and that pipe was causing the table to break.

**Comment by akgupta0777** [comment]
Seems great now. I am approving the changes and improvements.

**Comment by drph4nt0m** [comment]
Also added the same markdown to the wiki [here](https://github.com/drph4nt0m/avbot-charts/wiki/Playbooks)

**Pull request merged**